src/Microsoft.Identity.Client/Cache/Items/MsalItemWithAdditionalFields.cs

```diff
@@ -27,5 +26,30 @@ internal virtual JObject ToJObject()
 
             return json;
         }
+
+        internal void SetItemIfValueNotNull(JObject json, string key, JToken value)
+        {
+            bool shouldSetValue = true;
+
+            object asObj = value.ToObject<object>();
```

Comment:
what will this do to int/numeric fields?

Reply:
It boxes the int and we get the numeric value in asObj so this works fine.